### PR TITLE
Add support for ignoring times (`12:34`) `ignoreDigits` is on

### LINKED ISSUE #24: Ignore times in common formats (e.g. 2:41PM)
### Initial checklist

- [X] I read the [support docs](https://github.com/retextjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/retextjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/retextjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aretextjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

5.1.0

### Link to runnable example

https://peaceful-mclean-b30fef.netlify.app

### Steps to reproduce

1. Open the [runnable example](https://peaceful-mclean-b30fef.netlify.app).
2. In the text field, paste the following string: `Good morning, today is 2:41pm.`
3. In the right pane, you'll see that a warning is returned for the time of day.

## Expected behavior
**TL;DR**: I expect that the retext-spell plugin wouldn't return any warnings for times in common formats like `2:41pm`.

### Description
First, retext and unified as a whole is an _incredible project_. _Thank you_ for making and maintaining this.

I'm working on creating a JavaScript prose linter for anyone who writes copy in our UI. It's accessible via a browser or in a [Figma](http://figma.com) Plugin since it's primarily meant for product designers to use. [The prose linter](https://peaceful-mclean-b30fef.netlify.app) that I've created uses many existing retext plugins (like retext-spell) and several custom plugins for rules to help our teams adhere to our writing style guide. Since my company is a software monitoring tool, we often use times (e.g. 2:41pm) in our UI and it's very plausible that folks will attempt to use my plugin on text layers that contain times.

I'm taking advantage of the `ignoreDigits` option and it's working wonderfully! I was hoping to find another option that allows consumers of the project to control whether to ignore words that contain _any_ digits as opposed to _only_ digits (as the `ignoreDigits` option does).

### Affected runtime and version

node@14.15.5

### Affected package manager and version

npm@8.10.0

### Affected OS and version

macOS Monterey 12.2.1

### Build and bundle tools

webpack, Vite

### PR DESCRIPTION
**Pull request by danielgolden**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes
- Adds a check in the `ignoreDigits` option which will ignore any word that is a time (e.g. `2:41pm`)
- Adds a new test to ensure that times are ignored.
- Updates README to include the updated documentation for the `ignoreDigits` option.

---

Closes #24 and a step toward resolving newrelic/one-core-toolbox#38

### CODE REVIEW

**Review by wooorm — COMMENTED** [review]
Thanks for working on this! ✨

What you implemented is a bit different from what the original issue was (“times in common formats (e.g. 2:41PM)”).
There is a difference between:

- finite words, which also include numbers, and can accidentally include numbers (`w0rd`, because the `o` is next to `0`), which have a great solution already: the personal dictionary.
- numerical-like, such as times, which don’t have a good solution.

Are there other words you want to ignore? Did you consider the personal dictionary (or `ignore` if you have to)?

I was thinking more of adding `/^\d{1,2}:\d{2}(?:[ap]\.?m)?$/i`. Either under the existing `ignoreDigits` option. Or, if you like it, perhaps a new `ignoreTimes` option?

We could also add the resultions you show in a test case?

**Comment by danielgolden** [comment]
That's true, and a good point. There aren't really other words I want to ignore (times are my priority), they just seemed to fit in the same category as I was thinking about it (but I supposed this is exactly how scope creep happens haha).

> I was thinking more of adding /^\d{1,2}:\d{2}(?:[ap]\.?m)?$/i. Either under the existing ignoreDigits option

I really like this solution. Keeps the API simple and it fit's well the specific problem of ignoring times. I'll update the PR to reflect these changes.

Thanks for the feedback! :)

**Comment by danielgolden** [comment]
I'm not confident about the documentation updates. Is this a satisfactory way to document the ignoring of times? If so, awesome. If not, would love your opinion.

https://github.com/retextjs/retext-spell/blob/d71c22bf2917b28ade1157eb2c530e9a35002614/index.js#L18-L19
https://github.com/retextjs/retext-spell/blob/d71c22bf2917b28ade1157eb2c530e9a35002614/readme.md?plain=1#L112-L113

File: index.js [hunk 1]
```diff
@@ -286,6 +286,10 @@ function all(tree, file, config) {
    * @returns {boolean}
    */
   function irrelevant(word) {
-    return ignore.includes(word) || (ignoreDigits && /^\d+$/.test(word))
+    return (
+      ignore.includes(word) ||
+      (ignoreDigits && /^\d+$/.test(word)) ||
+      (ignoreDigits && /^\d{1,2}:\d{2}(?:[ap]\.?m)?$/i.test(word))
```
Comment (wooorm) [inline index.js]:
```suggestion
      (ignoreDigits && /^\d{1,2}:\d{2}(?:[ap]\.?m\.?)?$/i.test(word))
```

I think I missed this dot earlier that might be there?

File: test.js [hunk 2]
```diff
@@ -223,6 +223,17 @@ test('should ignore digits', (t) => {
     }, t.ifErr)
 })
 
+test('should ignore times', (t) => {
+  t.plan(1)
+
+  retext()
+    .use(retextSpell, enGb)
+    .process('2:41pm')
```
Comment (wooorm) [inline test.js]:
```suggestion
    .process('Let’s meet at 2:41pm.')
```

A more real example.

File: test.js [hunk 3]
```diff
@@ -223,6 +223,17 @@ test('should ignore digits', (t) => {
     }, t.ifErr)
 })
 
+test('should ignore times', (t) => {
+  t.plan(1)
```
Comment (wooorm) [inline test.js]:
```suggestion
  t.plan(2)
```

Let’s add another test.

File: test.js [hunk 4]
```diff
@@ -223,6 +223,17 @@ test('should ignore digits', (t) => {
...
+
+  retext()
+    .use(retextSpell, enGb)
+    .process('2:41pm')
+    .then((file) => {
+      check(t, file, [])
+    }, t.ifErr)
```
Comment (wooorm) [inline test.js:234]:
```suggestion
    }, t.ifErr)

  retext()
    .use(retextSpell, enGb)
    .process("On my way! ETA 11:50!")
    .then((file) => {
      check(t, file, []);
    }, t.ifErr);
```

Another case for just a time

**Review by wooorm — COMMENTED** [review]
Docs look good!

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks, released in [`5.3.0`](https://github.com/retextjs/retext-spell/releases/tag/5.3.0)!

**Comment by danielgolden** [comment]
Awesome! Thanks so much :)
